src/mlpack/methods/ann/init_rules/const_init.hpp

```diff
@@ -1,33 +1,37 @@
 /**
- * @file zero_init.hpp
- * @author Marcus Edel
+ * @file const_init.hpp
+ * @author Sumedh Ghaisas
  *
  * Intialization rule for the neural networks. This simple initialization is
- * performed by assigning a zero matrix to the weight matrix.
+ * performed by assigning a matrix of all constant values to the weight
+ * matrix.
  *
  * mlpack is free software; you may redistribute it and/or modify it under the
  * terms of the 3-clause BSD license.  You should have received a copy of the
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
-#ifndef MLPACK_METHODS_ANN_INIT_RULES_ZERO_INIT_HPP
-#define MLPACK_METHODS_ANN_INIT_RULES_ZERO_INIT_HPP
+#ifndef MLPACK_METHODS_ANN_INIT_RULES_CONST_INIT_HPP
+#define MLPACK_METHODS_ANN_INIT_RULES_CONST_INIT_HPP
 
 #include <mlpack/prereqs.hpp>
 
 namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
 /**
- * This class is used to initialize randomly the weight matrix.
+ * This class is used to initialize weight matrix with constant values.
+ *
+ * @tparam initConst Const value to initialize.
  */
-class ZeroInitialization
+template<size_t initConst>
```

Comment:
Agreed, I think the time savings from using a compile-time constant aren't going to be noticeable in this case, and it's pretty restrictive to disallow non-size_t values.